tests/trainer/test_trainer.py

```diff
@@ -889,6 +889,21 @@ def on_after_backward(self):
     assert not torch.isfinite(params).all()
 
 
+def test_detect_anomaly_nan(tmpdir):
+    class CurrentModel(BoringModel):
+        def forward(self, x):
+            x /= 0
```

Comment:
why not set NaN directly?
at least add comments that we user zero division to get NaN of the same size or original array